File: CPE-1star/UVA10922-2 the 9s .py
while True:
	try:
		N=int(input())
		if N==0:
			break
		if N%9!=0:
			print("%d is not a multiple of 9."%(N))
		else:
			temp=0
			subN=N
			degree=1
			while subN%9==0:
				while subN>0:
					temp+=subN%10
					subN=subN//10
				if temp%9==0:
					degree+=1
				subN=temp
			print("%d is a multiple of 9 and has 9-degree %d."%(N,degree))
	except:
		break

File: CPE-1star/UVA10922-2 the 9s .cpp
#include<iostream>
using namespace std;
int main(){
    long long n;
    while(cin>>n){
        if(n==0)
            break;
        if(n%9!=0)
            cout<<n<<" is not a multiple of 9."<<"\n";
        else{
            int temp=0,subn=n,degree=0;
            while(subn%9==0){
                while(subn>0){
                    temp+=subn%10;
                    subn=subn/10;
                }
                if(temp%9==0)
                    degree+=1;
                subn=temp;
                temp=0;
                if(subn==9)
                    break;
            }
            cout<<n<<" is a multiple of 9 and has 9-degree "<<degree<<"."<<'\n';
        }
    }
    return 0;
}